connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -592,18 +595,45 @@ private static boolean fetchBooleanProperty(ApplicationInfo appInfo, String prop
         return (Boolean) booleanPropertyObject;
     }
 
+    // This operation may (and will) cause UI delay for the user. This is done on purpose.
+    // Cleaning of invalid idp configuration must be ensured.
     private static void validateWellKnownConfigIdp() {
-        String wellKnownConfigIssuer = null;
+        // We expect the current current selected id provider to be the issuer
+        final String expectedIssuer = ConnectUrlHelper.getConnectApiUrl(idProvider, useStaging) + ConnectUrlHelper.OAUTH_PATH;
+
+        if (connectStore != null) {
+            IdToken idToken = connectStore.getIdToken();
+            if (idToken != null) {
+                final ReadOnlyJWTClaimsSet idTokenClaimsSet;
+
+                try {
+                    final SignedJWT signedJwt = SignedJWT.parse(idToken.getSerializedSignedJwt());
+                    idTokenClaimsSet = signedJwt.getJWTClaimsSet();
+                } catch (final ParseException e) {
+                    wellKnownConfig = null;
+                    lastSeenWellKnownConfigStore.clearSynchronously();
+                    connectStore.clearSynchronously();
+                    return;
+                }
+
+                final String issuer = idTokenClaimsSet.getIssuer();
+                if (!expectedIssuer.equals(issuer)) {
+                    wellKnownConfig = null;
+                    lastSeenWellKnownConfigStore.clearSynchronously();
+                    connectStore.clearSynchronously();
+                    return;
+                }
+                return;
+            }
+        }
+
         if (wellKnownConfig != null) {
-            wellKnownConfigIssuer = wellKnownConfig.getIssuer();
-        }
-        String idpIssuer = ConnectUrlHelper.getConnectApiUrl(idProvider, useStaging) + ConnectUrlHelper.OAUTH_PATH;
-        if (!idpIssuer.equals(wellKnownConfigIssuer)) {
-            wellKnownConfig = null;
-            // This operation may (and will) cause UI delay for the user. This is done on purpose.
-            // Cleaning of invalid idp configuration must be ensured.
-            lastSeenWellKnownConfigStore.clearSynchronously();
-            connectStore.clearSynchronously();
+            String wellKnownConfigIssuer = wellKnownConfig.getIssuer();
+            if (!expectedIssuer.equals(wellKnownConfigIssuer)) {
+                wellKnownConfig = null;
+                lastSeenWellKnownConfigStore.clearSynchronously();
+                connectStore.clearSynchronously();
```

Comment:
We are testing if connectStore is not null higher up in this method. Could connectStore be null here?

Reply:
It shouldn't, however it's safer to keep a check to prevent possible issues in future.